analyzer_test.go

```diff
@@ -198,263 +122,111 @@ func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
 					})
-				})
-			})
-
-			when("skip-layers is true", func() {
-				it.Before(func() {
-					analyzer.SkipLayers = true
-				})
-
-				it("should return the analyzed metadata", func() {
-					md, err := analyzer.Analyze(image, testCache)
-					h.AssertNil(t, err)
-
-					h.AssertEq(t, md.Image.Reference, "s0m3D1g3sT")
-					h.AssertEq(t, md.Metadata, appImageMetadata)
-				})
 
-				it("does not write buildpack layer metadata", func() {
-					_, err := analyzer.Analyze(image, testCache)
-					h.AssertNil(t, err)
+					when("skip-layers is true", func() {
+						it.Before(func() {
+							analyzer.SkipLayers = true
+							mockLayerAnalyzer.EXPECT().RetrieveMetadataFrom(testCache)
+							mockLayerAnalyzer.EXPECT().Analyze(analyzer.Buildpacks, analyzer.SkipLayers, gomock.Any(), gomock.Any())
+						})
 
-					files, err := ioutil.ReadDir(layerDir)
-					h.AssertNil(t, err)
-					h.AssertEq(t, len(files), 2)
+						it("should return the analyzed metadata", func() {
+							md, err := analyzer.Analyze(image, testCache)
+							h.AssertNil(t, err)
 
-					files, err = ioutil.ReadDir(filepath.Join(layerDir, "metadata.buildpack"))
-					h.AssertNil(t, err)
-					//expect 1 file b/c of store.toml
-					h.AssertEq(t, len(files), 1)
-
-					files, err = ioutil.ReadDir(filepath.Join(layerDir, "no.cache.buildpack"))
-					h.AssertNil(t, err)
-					//expect 1 file b/c of store.toml
-					h.AssertEq(t, len(files), 1)
-				})
-
-				it("restores each store metadata", func() {
-					_, err := analyzer.Analyze(image, testCache)
-					h.AssertNil(t, err)
-					for _, data := range []struct{ name, want string }{
-						// store.toml files.
-						{"metadata.buildpack/store.toml", "[metadata]\n  [metadata.metadata-buildpack-store-data]\n    store-key = \"store-val\""},
-						{"no.cache.buildpack/store.toml", "[metadata]\n  [metadata.no-cache-buildpack-store-data]\n    store-key = \"store-val\""},
-					} {
-						got := h.MustReadFile(t, filepath.Join(layerDir, data.name))
-						h.AssertStringContains(t, string(got), data.want)
-					}
+							h.AssertEq(t, md.Image.Reference, "s0m3D1g3sT")
+							h.AssertEq(t, md.Metadata, appImageMetadata)
+						})
+					})
 				})
 			})
 		})
 
-		when("image is not found", func() {
+		when("platform API < 0.6", func() {
 			it.Before(func() {
-				h.AssertNil(t, image.Delete())
+				analyzer.PlatformAPI = api.MustParse("0.5")
 			})
 
-			when("cache exists", func() {
+			when("image is not found", func() {
 				it.Before(func() {
-					metadata := h.MustReadFile(t, filepath.Join("testdata", "analyzer", "cache_metadata.json"))
-					var cacheMetadata platform.CacheMetadata
-					h.AssertNil(t, json.Unmarshal(metadata, &cacheMetadata))
-					h.AssertNil(t, testCache.SetMetadata(cacheMetadata))
-					h.AssertNil(t, testCache.Commit())
-
-					analyzer.Buildpacks = append(analyzer.Buildpacks, buildpack.GroupBuildpack{ID: "escaped/buildpack/id"})
-				})
-
-				it("restores cache=true layer metadata", func() {
```

Comment:
Where did this (and the following two) move to?

Reply:
> restores app and cache layer metadata
That test covers this file already